plugins/trace_api_plugin/trace_api_plugin.cpp

```diff
@@ -0,0 +1,229 @@
+      auto cfg_options = cfg.add_options();
+      cfg_options("trace-rpc-abi", bpo::value<vector<string>>()->composing(),
+                  "ABIs used when decoding trace RPC responses, specified as \"Key=Value\" pairs in the form <account-name>=<abi-def>\n"
+                  "Where <abi-def> can be:\n"
+                  "   a valid JSON-encoded ABI as a string\n"
+                  "   an absolute path to a file containing a valid JSON-encoded ABI\n"
+                  "   a relative path from `data-dir` to a file containing a valid JSON-encoded ABI\n"
+                  );
+   }
+
+   void plugin_initialize(const appbase::variables_map& options) {
+      data_handler = std::make_shared<abi_data_handler>([](const exception_with_context& e){
+         log_exception(e, fc::log_level::debug);
+      });
+
+      if( options.count("trace-rpc-abi") ) {
+         const std::vector<std::string> key_value_pairs = options["trace-rpc-abi"].as<std::vector<std::string>>();
+         for (const auto& entry : key_value_pairs) {
+            try {
+               auto kv = parse_kv_pairs(entry);
+               auto account = chain::name(kv.first);
+               auto abi = abi_def_from_file_or_str(kv.second);
+               data_handler->add_abi(account, abi);
+            } catch (...) {
+               elog("Malformed trace-rpc-abi provider: \"${val}\", ignoring!", ("val", entry));
+            }
+         }
+      }
+   }
+
+   void plugin_startup() {
+   }
+
+   void plugin_shutdown() {
+   }
+
+   std::shared_ptr<trace_api_common_impl> common;
+   std::shared_ptr<abi_data_handler> data_handler;
+};
+
+struct trace_api_plugin_impl {
+   trace_api_plugin_impl( const std::shared_ptr<trace_api_common_impl>& common )
+   :common(common) {}
+
+   static void set_program_options(appbase::options_description& cli, appbase::options_description& cfg) {
+      auto cfg_options = cfg.add_options();
+      cfg_options("minimum-irreversible-trace-history", bpo::value<std::string>()->default_value("-1"),
+                  "the minimum amount of history, as defined by time, this node will keep after it becomes irreversible\n"
+                  "this value can be specified as a number of seconds or using the following suffixes:\n"
+                  "  \"s\" : the value is in seconds\n"
+                  "  \"m\" : the value is in minutes\n"
+                  "  \"h\" : the value is in hours\n"
+                  "  \"d\" : the value is in days\n"
+                  "  \"w\" : the value is in weeks\n"
+                  "a value of \"-1\" will disable automatic maintenance of the trace slice files\n"
+                  );
+   }
+
+   void plugin_initialize(const appbase::variables_map& options) {
+      if( options.count("minimum-irreversible-trace-history") ) {
+         auto value = options.at("minimum-irreversible-trace-history").as<std::string>();
+         try {
+            minimum_irreversible_trace_history = parse_microseconds(value);
```

Comment:
`parse_microseconds` is nice. However, no other existing option supports this capability. I think this needs to be lifted out into the main code base and used for all options that take microseconds (or really any time) or removed for now.